framework/src/meshgenerators/DistributedRectilinearMeshGenerator.C

```diff
@@ -216,14 +216,29 @@ DistributedRectilinearMeshGenerator::addElement<Edge2>(const dof_id_type nx,
 
   auto node_offset = elem_id;
 
-  auto node0_ptr = mesh.add_point(Point(static_cast<Real>(node_offset) / nx, 0, 0), node_offset);
-  node0_ptr->set_unique_id() = node_offset;
-  node0_ptr->processor_id() = pid;
+  Node * node0_ptr = mesh.query_node_ptr(node_offset);
```

Comment:
Does this query involve communication?

Reply:
No; this the "see if a object with the requested id exists" API, purely local.